runtime/metricsview/ast.go

```diff
@@ -1106,6 +1110,18 @@ func (a *AST) addTimeComparisonMeasure(n *SelectNode, m *runtimev1.MetricsViewSp
 	if err != nil {
 		return err
 	}
+
+	// This is mainly required for getting the correct data type for druid as it wraps all measures in ANY_VALUE in rewriteDruidGroups.
```

Comment:
Another way is to just solve this for druid by looking for presence of time column computation pattern in `rewriteDruidGroups` and not wrapping that measure but that seems fragile.